src/lib/cosmos.decorator.tsx

```diff
@@ -0,0 +1,215 @@
+import { Web3Provider } from '@ethersproject/providers'
+import { UnsupportedChainIdError, useWeb3React, Web3ReactProvider } from '@web3-react/core'
+import {
+  InjectedConnector,
+  NoEthereumProviderError,
+  UserRejectedRequestError as UserRejectedRequestErrorInjected,
+} from '@web3-react/injected-connector'
+import { NetworkConnector } from '@web3-react/network-connector'
+import { useEffect, useState } from 'react'
+
+const RPC_URLS: { [chainId: number]: string } = {
+  1: `https://mainnet.infura.io/v3/4bf032f2d38a4ed6bb975b80d6340847`,
+  4: `https://rinkeby.infura.io/v3/4bf032f2d38a4ed6bb975b80d6340847`,
+}
+
+export const injected = new InjectedConnector({ supportedChainIds: [1, 3, 4, 5, 42] })
+
+export const network = new NetworkConnector({
+  urls: { 1: RPC_URLS[1], 4: RPC_URLS[4] },
+  defaultChainId: 1,
+})
+
+export function useEagerConnect() {
```

Comment:
Use the existing logic in the repo.
Instead of copying this logic over to `lib`, I think it'd be better to just expand the interface logic so that it is usable by `lib`, and worry about completely porting in only when widgets are ready to replace the interface functionality.